### PR TITLE
add landing page for node-red learning resources docs

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Description

Added the landing page for Node-RED learning resources and set the order of nav items.

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Comment by sumitshinde-84** [comment]
The test is failing because the URL needs to be updated where we have interlinked. I will update it once the final decision on the learning resources URL is made.

**Comment by Yndira-E** [comment]
Hi @sumitshinde-84 the idea wasn't to move any of the content under /node-red/learn, just to create a learn.md file and add it in the left-nav as another item, it should be the first one, but everything else remains where it was.

**Review by Yndira-E — APPROVED** [review]

File: src/node-red/learn.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  order: 1
+meta:
+   title: Node-RED Learning Resources
+   description: Explore this comprehensive set of Node-RED learning resources, including how to set up Node-RED on various hardware platforms, its basic terminologies, integration of different databases, and more.
+   keywords: node-red, node red, tutorial, learning resources, guide, node-red basics, node-red foundation. 
+---
+
+# Node-RED Learning Resources
+
+[Node-RED](/node-red/) is a low-code visual platform used for building IoT applications. This platform allows the integration of different hardware devices, API services, and technologies makes it exceptionally suited for industrial IoT applications.
```
Comment (Hasmin-AC) [inline src/node-red/learn.md:13]:
```suggestion
[Node-RED](/node-red/), the low-code visual programming platform, is becoming increasingly popular for optimizing and automating industrial processes. Enabling the integration of different hardware devices, API services, and technologies, it's exceptionally suited for a variety of industrial IoT applications.
```

File: src/node-red/learn.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+This documentation will help you get started with Node-RED, understand Node-RED Core Nodes, familiarize yourself with terminologies, and provide tutorials on integrating Node-RED with different databases, technologies, and protocols. Additionally, it covers how to run Node-RED on various hardware platforms.
```
Comment (Hasmin-AC) [inline src/node-red/learn.md:15]:
```suggestion
Here you will find all the resources you need to help you get started with Node-RED, understand Node-RED Core Nodes, familiarize yourself with terminologies, and tutorials on integrating Node-RED with different databases, technologies, and protocols. Additionally, we cover how to run Node-RED on various hardware platforms. 
```

File: src/node-red/learn.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+*Note: Currently, each section covers a few topics, but more content is expected to be added in the future.*
```
Comment (Hasmin-AC) [inline src/node-red/learn.md:17]:
```suggestion
*Note: We will be continuously adding new topics of content so be sure to bookmark and check back in.*
```

File: src/node-red/learn.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+## Common Terminology
+
+This document contains some of the most common terminologies used in Node-RED communities and our resources. By learning these common terms, you can confidently navigate through most Node-RED-related resources and participate in basic Node-RED forum discussions.
```
Comment (Hasmin-AC) [inline src/node-red/learn.md:21]:
```suggestion
A good starting point is understanding the common terms used in Node-RED. You should be able to confidently navigate through most Node-RED-related resources and participate in Node-RED forum discussions.
```

File: src/node-red/learn.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+[Node-RED Common Terminologies](/node-red/terminology/)
+
+## Core Nodes
+
+This section includes detailed documentation for each Node-RED core node, from configuration to use in real-world applications with examples.
+
+[Core Nodes Documentation](/node-red/core-nodes/)
```
Comment (Hasmin-AC) [inline src/node-red/learn.md:29]:
```suggestion
[Core Nodes](/node-red/core-nodes/)
```

File: src/node-red/learn.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+## Hardware Devices
+
+This section of our documentation includes tutorials on setting up Node-RED on different hardware devices, tutorials covers configuration to practicle demostration along with troubleshooting tips.
```
Comment (Hasmin-AC) [inline src/node-red/learn.md:33]:
```suggestion
In this section, you will find tutorials on setting up Node-RED on different hardware devices. Each tutorial covers configuration, practical demos and troubleshooting tips.
```

File: src/node-red/learn.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+## Peripheral Devices
+
+This section includes step-by-step documentation on integrating Node-RED with various peripheral devices such as webcams, barcode scanners, printer, mic and more.
+
+[Integrating Node-RED with Peripheral Devices](/node-red/peripheral/)
+
+## Databases
+
+This section of our documentation includes tutorials on how to use different databases with Node-RED. Each document provides step-by-step instructions along with practical examples. This section covers various types of databases such as SQL, NoSQL, and time-series databases.
```
Comment (Hasmin-AC) [inline src/node-red/learn.md:45]:
```suggestion
This section includes tutorials on how to use different databases with Node-RED. Each tutorial provides step-by-step instructions along with practical examples. We cover a variety of databases such as SQL, NoSQL, and time-series databases.
```

File: src/node-red/learn.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+[Node-RED Database Integration Guides](/node-red/database/)
+
+## Communication Protocols
+
+This section of our documentation includes comprehensive tutorials on how to use different protocols with Node-RED for building applications, communicating with edge devices, and more. Each tutorial provides step-by-step instructions along with practical demonstrations.
```
Comment (Hasmin-AC) [inline src/node-red/learn.md:51]:
```suggestion
This section includes comprehensive tutorials on how to use different protocols with Node-RED for building applications, communicating with edge devices, and more. Each tutorial provides step-by-step instructions along with practical demos.
```

File: src/node-red/learn.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+[Using Different Protocols for Building Applications with Node-RED](/node-red/protocol/)
+
+## Integration Technologies
+
+This section of our documentation includes tutorials on how to utilize various technologies with Node-RED, such as implementing webhooks for task automation, creating REST APIs, GraphQL integrations, and more.
```
Comment (Hasmin-AC) [inline src/node-red/learn.md:57]:
```suggestion
In this section you will find tutorials on how to utilize various technologies with Node-RED, such as implementing webhooks for task automation, creating REST APIs, GraphQL integrations, and more.
```

File: src/node-red/learn.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+[Using Different Technologies for Building Applications with Node-RED](/node-red/integration-technologies/)
+
+## Notification Services
+
+This section of the documentation covers how to integrate various notification services with Node-RED to send real-time alerts and notifications. It includes step-by-step guides along with troubleshooting tips.
```
Comment (Hasmin-AC) [inline src/node-red/learn.md:63]:
```suggestion
In this section, we cover how to integrate various notification services with Node-RED to send real-time alerts and notifications. It includes step-by-step guides along with troubleshooting tips.
```

**Review by Hasmin-AC — APPROVED** [review]

**Pull request merged**